### DISCUSSION TITLE
请问如何仅提取 zotero“extra” 栏中的数字到笔记属性模板中

### DISCUSSION THREAD

**Discussion by Litt7e** [aidenlx/obsidian-zotero#284] [Help]
我一般会在zoter的“extra” 栏中对文献进行评级，如"rate: 5"，但是在extra栏中也可能会有自带的其他注释。请问如何仅提取“rate: ”后面的数字到笔记属性模板中呢

**Comment by Silence2540** [comment]
`<%= (it.extra.first().match(/rate:\s*(.*?)(\n|$)/s) || [])[1]?.trim() %>`

**Comment by aidenlx** [comment]
v2 会直接解析 Extra 字段，不用再写正则了：

```liquid
rate: {{ zt.extra.fields["rate"] }}
```

`zt.extra.fields` 按键名取第一个值，Extra 里其他自带的注释不会被带进来；没有 `rate` 这个键时读出来是空值，模板也不会报错。

v2 beta 现已发布：https://zotlit.aidenlx.site/docs/install-zotlit
迁移指南：https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
